src/server.rs

```diff
@@ -83,14 +88,36 @@ impl Conmon for ConmonServerImpl {
 
 // Use the single threaded runtime to save rss memory
 #[tokio::main(flavor = "current_thread")]
-async fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let addr = "[::1]:50051".parse()?;
+async fn main() -> Result<(), Error> {
+    let (shutdown_tx, shutdown_rx) = tokio::sync::oneshot::channel();
+
     let server = ConmonServerImpl::new()?;
 
+    let sigterm_handler = tokio::spawn(start_sigterm_handler(shutdown_tx));
+    let grpc_backend = tokio::spawn(start_grpc_backend(server, shutdown_rx));
+
+    let _ = tokio::join!(sigterm_handler, grpc_backend);
+    Ok(())
+}
+
+async fn start_sigterm_handler(shutdown_tx: oneshot::Sender<()>) {
+    let mut sigterm = signal(SignalKind::terminate()).unwrap();
+    sigterm.recv().await;
+    info!("received SIGTERM");
+    let _ = shutdown_tx.send(());
+}
```

Comment:
Let's return a result here to avoid the `.unwrap()` call:
```suggestion
async fn start_sigterm_handler(shutdown_tx: oneshot::Sender<()>) -> Result<()> {
    let mut sigterm = signal(SignalKind::terminate())?;
    sigterm.recv().await;
    info!("received SIGTERM");
    let _ = shutdown_tx.send(());
    Ok(())
}

```